routers/user/oauth.go

```diff
@@ -193,6 +194,40 @@ func newAccessTokenResponse(grant *models.OAuth2Grant, clientSecret string) (*Ac
 	}, nil
 }
 
+type userInfoResponse struct {
+	Sub      string `json:"sub"`
+	Name     string `json:"name"`
+	Username string `json:"preferred_username"`
+	Email    string `json:"email"`
+	Picture  string `json:"picture"`
+}
+
+// InfoOAuth manages request for userinfo endpoint
+func InfoOAuth(ctx *context.Context) {
+	header := ctx.Req.Header.Get("Authorization")
+	auths := strings.Fields(header)
+	if len(auths) != 2 || auths[0] != "Bearer" {
+		ctx.HandleText(http.StatusUnauthorized, "no valid auth token authorization")
+		return
+	}
+	uid := sso.CheckOAuthAccessToken(auths[1])
```

Comment:
if uid == 0, we should still give an error.

Reply:
Good point, will add this afternoon.

Reply:
If we can check `uid == 0` and return, it's a better control flow on Golang.

And it should not be a server error but an invalid token.

Reply:
Ah ok, so you mean a
```
if uid == 0 {
  handleAccessTokenError()
  return
}
```
before the 
`if uid != 0 {`
?
Or should I remove the If statement uid != 0 before the next block?

Sry, but I'm new to golang, so I am not so firm with control flows on Golang. (Coming from Python and some other languages)

Reply:
Ok, just found RFC6750 for invalid bearer token usage.
Will implement a new handleBearerTokenError() function which should be used when there is an error at the UserInfo Endpoint according to the specs.

Reply:
Implemented handleBearerTokenError and the BearerTokenError according to RFC 6750.
I didn't invalidate the token inside gitea, because the access token for itself is valid at the client and in RFC 6750 stands nothing about invalidation of any accesstoken, it should only respond with an invalid_token error and status code 401.